presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -1507,6 +1508,67 @@ private static void validateSphericalType(String function, OGCGeometry geometry,
         }
     }
 
+    @SqlNullable
+    @Description("Returns the area of a geometry on the Earth's surface")
+    @ScalarFunction("ST_Area")
+    @SqlType(DOUBLE)
+    public static Double stSphericalArea(@SqlType(SPHERICAL_GEOGRAPHY_TYPE_NAME) Slice input)
+    {
+        OGCGeometry geometry = deserialize(input);
+        if (geometry.isEmpty()) {
+            return null;
+        }
+
+        validateSphericalType("ST_Area", geometry, EnumSet.of(POLYGON));
+        Polygon polygon = (Polygon) geometry.getEsriGeometry();
+
+        return sphericalPolygonArea(polygon);
+    }
+
+    private static double sphericalPolygonArea(Polygon polygon) {
+
+        int nbPoints = polygon.getPointCount();
+
+        // Validate input
+        if (nbPoints < 3) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, format("Can't calculate the area of a polygon with less than 3 vertices. Input has %d", nbPoints));
+        }
+
+
+        for(int i = 0 ; i < nbPoints ; i++) {
+            Point p = polygon.getPoint(i);
+            double latitude = p.getY();
+            double longitude = p.getX();
+            checkLatitude(latitude);
```

Comment:
These checks are redundant. They already ran as part of to_spherical_geography function call. See GeoFunctions#toSphericalGeography